### ISSUE TITLE
[Google Drive] No results returned. Indexing files for the first time, every time.

### ISSUE THREAD

**Issue by christianmagill** [raycast/extensions#4398]
### Extension

https://www.raycast.com/vishaltelangre/google-drive

### Description

I've set up my Google Drive path, and every time I run the File Search command it shows the files being indexed. I wait until the indexing is no longer running, but still no results are returned for files I know exist.

### Steps To Reproduce

Using the latest Raycast (1.46.0), extension and Mac OSX Ventura 13.1 (22C65)

**Comment by daniel-stoneuk** [comment]
I believe Google Drive has switched to the [FileProvider](https://developer.apple.com/documentation/fileprovider) API which could mean that this extension no longer works.

We should investigate this in case it is worth temporarily removing it from the store until fixed. As a workaround, there is now a Google Drive hotkey <kbd>⌥ Option</kbd> + <kbd>⌘ Command</kbd> + <kbd>G</kbd> to search for files and it's pretty fast!

**Comment by vishaltelangre** [comment]
> I believe Google Drive has switched to the [FileProvider](https://developer.apple.com/documentation/fileprovider) API which could mean that this extension no longer works.

It still works for me (I am on Ventura 13.1, too). This is how I have set the Drive root path in Raycast.

![image](https://user-images.githubusercontent.com/876195/214308307-5204a44b-f737-4f2b-b43e-4a8de3109b98.png)

This is where the Google Drive appears in the Finder app.

![image](https://user-images.githubusercontent.com/876195/214308796-67a42096-1851-47ca-9c79-db3d700bbda0.png)

I am able to index files (pretty fast as usual, though I don't have many files in my Google Drive).

I believe something else might be causing in OP's case that is making it to index forever (because they don't seem to have reported any error).

But like @daniel-stoneuk said, using the native Google Drive app itself might be a good idea now, given it can be triggered with a global shortcut.

**Comment by christianmagill** [comment]
Resetting the root path let me see the indexing taking place again. It gets to about 60%, after several minutes, and then just stops. I do however have a large number of files.

**Comment by ahoereth** [comment]
I think this issue still applies, the extension is also basically unusable for me with over 100k files in drive. 

Thing is, I think the problem is somewhat outside of the reach of the Raycast extensions. During indexing the extension basically globs for all files in the provided directory and adds them to a sql.js database one by one. To figure out whether the slow part is listing files or adding them to the database, I ran `find ~/Google\ Drive/ | fzf`: It is also very very slow to list files, strongly hinting at the former being the problem. Independent of that I implemented bulk inserts to the extension (100 files at a time), but that did not increase the indexing speed enough for the extension to be useable for me.

Not sure if @vishaltelangre or @daniel-stoneuk have other ideas? I will probably switch to the Google Workspace extension, which has fast online search. It does not have offline search and currently does not provide a "open file on local filesystem" option, but the latter can probably be added using some heuristics. Also, Raycast's own file search command mostly works well enough for me for searching drive, can't really say in which situation it fails me yet.
